File: web/src/api/nginx_domain_rule.js
import service from '@/utils/request'

// @Tags DomainRule
// @Summary 创建DomainRule
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DomainRule true "创建DomainRule"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /Rule/createDomainRule [post]
export const createDomainRule = (data) => {
     return service({
         url: "/Rule/createDomainRule",
         method: 'post',
         data
     })
 }


// @Tags DomainRule
// @Summary 删除DomainRule
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DomainRule true "删除DomainRule"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /Rule/deleteDomainRule [delete]
 export const deleteDomainRule = (data) => {
     return service({
         url: "/Rule/deleteDomainRule",
         method: 'delete',
         data
     })
 }

// @Tags DomainRule
// @Summary 删除DomainRule
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除DomainRule"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /Rule/deleteDomainRule [delete]
 export const deleteDomainRuleByIds = (data) => {
     return service({
         url: "/Rule/deleteDomainRuleByIds",
         method: 'delete',
         data
     })
 }

// @Tags DomainRule
// @Summary 更新DomainRule
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DomainRule true "更新DomainRule"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /Rule/updateDomainRule [put]
 export const updateDomainRule = (data) => {
     return service({
         url: "/Rule/updateDomainRule",
         method: 'put',
         data
     })
 }


// @Tags DomainRule
// @Summary 用id查询DomainRule
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DomainRule true "用id查询DomainRule"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /Rule/findDomainRule [get]
 export const findDomainRule = (params) => {
     return service({
         url: "/Rule/findDomainRule",
         method: 'get',
         params
     })
 }


// @Tags DomainRule
// @Summary 分页获取DomainRule列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.PageInfo true "分页获取DomainRule列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /Rule/getDomainRuleList [get]
 export const getDomainRuleList = (params) => {
     return service({
         url: "/Rule/getDomainRuleList",
         method: 'get',
         params
     })
 }

File: server/router/nginx_domain_rule.go
package router

import (
	"nginx-web/api/v1"
	"nginx-web/middleware"
	"github.com/gin-gonic/gin"
)

func InitDomainRuleRouter(Router *gin.RouterGroup) {
	DomainRuleRouter := Router.Group("Rule").Use(middleware.JWTAuth()).Use(middleware.CasbinHandler()).Use(middleware.OperationRecord())
	{
		DomainRuleRouter.POST("createDomainRule", v1.CreateDomainRule)   // 新建DomainRule
		DomainRuleRouter.DELETE("deleteDomainRule", v1.DeleteDomainRule) // 删除DomainRule
		DomainRuleRouter.DELETE("deleteDomainRuleByIds", v1.DeleteDomainRuleByIds) // 批量删除DomainRule
		DomainRuleRouter.PUT("updateDomainRule", v1.UpdateDomainRule)    // 更新DomainRule
		DomainRuleRouter.GET("findDomainRule", v1.FindDomainRule)        // 根据ID获取DomainRule
		DomainRuleRouter.GET("getDomainRuleList", v1.GetDomainRuleList)  // 获取DomainRule列表
	}
}
